accounts/views.py

```
@@ -0,0 +1,135 @@
+            }
+        )
+
+    def post(self, request):
+        password = CustomUser.objects.make_random_password(10)
+        # form1 = CustomUserCreationForm(request.POST)
+        form1 = CustomUserCreationFormFirst2(request.POST)
+
+        if form1.is_valid():
+            form1.save()
+            username = form1.instance.username
+            email = form1.instance.email
+            protocol = request.get_host
+            domain = 'accounts/password_reset/'
+            res = {
+                'username': username,
+                'email': email,
+                'protocol': protocol,
+                'domain': domain,
+            }
+            body = render_to_string('user_registration/profile_reset.txt', res)
+            send_mail(
+                'Hello new user',
+                body,
+                'FDBteam@sda.com',
+                [email],
+                fail_silently=False,
+            )
+
+        return redirect('accounts:list-profile')
+
+
+class ProfileUpdateView(views.View):
+
+    def get(self, request, pk):
+        user = get_object_or_404(CustomUser, pk=pk)
+        profile = get_object_or_404(CustomUserProfile, pk=pk)
+
+        user_form = CustomUserChangeForm(instance=user)
+        profile_form = CustomUserProfileChangeForm(instance=profile)
+
+        return render(
+            request,
+            'accounts/profile_update.html',
+            {'user_form': user_form,
+             'profile_form': profile_form},
+        )
+
+    def post(self, request, pk):
+        user = get_object_or_404(CustomUser, pk=pk)
+        profile = get_object_or_404(CustomUserProfile, pk=pk)
+
+        user_form = CustomUserChangeForm(request.POST, instance=user)
+        profile_form = CustomUserProfileChangeForm(request.POST, request.FILES, instance=profile)
+
+        try:
+            old_image = profile.upload.path
+            if os.path.exists(old_image):
+                os.remove(old_image)
+        except:
```

UWAGA! Tak nie robimy. Jeżeli już bardzo chcemy coś wyciszyć to wyciszamy bardzo konkretny wyjątek. Ale najlepiej w ogóle nie wyciszać, tylko zostawiać jakiś ślad w logach (w pliku, może w oddzielnej tabelce z błędami w bazie), z informacją co poszło nie tak.
Jeżeli nie wskażemy konkretnego wyjątku to except: pass spowoduje, że nie dowiemy się o niczym co poszłoby nie tak przy usuwaniu pliku. I nigdzie nie znajdziemy śladu co poszło nie tak. To tak jakbyśmy przed rozpoczęciem debugowania zawiązywali sobie oczy przepaską. 
https://stackoverflow.com/questions/21553327/why-is-except-pass-a-bad-programming-practice